src/main/java/seedu/address/model/person/Address.java

```diff
@@ -44,9 +50,19 @@ public String toString() {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && value.equals(((Address) other).value)); // state check
+        if (other == this) {
+            return true;
+        } else if (other instanceof Address) {
+            if (value == null) {
+                return Objects.isNull(((Address) other).value);
```

Comment:
nitpicking, but might be cleaner to type-cast other after line 55 to make the code neater